src/Networking/NexusMods.Networking.GOG/Client.cs

```diff
@@ -0,0 +1,390 @@
+    {
+        var authQuery = new Dictionary<string, string?>()
+        {
+            { "client_id", ClientId },
+            { "client_secret", ClientSecret },
+            { "response_type", "code" },
+            { "redirect_uri", RedirectUrl.ToString() },
+        };
+
+        var urlTask = _authUrls.Reader.ReadAsync(token).AsTask();
+        
+        await _osInterop.OpenUrl(new Uri(QueryHelpers.AddQueryString(AuthorizationUri.ToString(), authQuery)), cancellationToken: token);
+
+        if (Task.WaitAny([urlTask], token) != 0)
+        {
+            _logger.LogWarning("The OAuth login request was cancelled.");
+            return;
+        }
+
+        var code = urlTask.Result.Code;
+        
+        // Request the token
+        var tokenQuery = new Dictionary<string, string?>
+        {
+            { "client_id", ClientId },
+            { "client_secret", ClientSecret },
+            { "code", code },
+            { "grant_type", "authorization_code" },
+            { "redirect_uri", RedirectUrl.ToString() },
+        };
+
+        var uri = new Uri(QueryHelpers.AddQueryString(TokenUri.ToString(), tokenQuery));
+
+        var tokenResponse = await _client.GetFromJsonAsync<TokenResponse>(uri, token);
+
+        if (tokenResponse == null)
+        {
+            _logger.LogError("The OAuth login request did not return a token.");
+            return;
+        }
+
+        // Save the login information
+        using var tx = _connection.BeginTransaction();
+        var e = tx.TempId();
+        if (TryGetAuthInfo(out var found))
+            e = found.Id;
+
+        tx.Add(e, AuthInfo.AccessToken, tokenResponse.AccessToken);
+        tx.Add(e, AuthInfo.RefreshToken, tokenResponse.RefreshToken);
+        tx.Add(e, AuthInfo.ExpiresAt, DateTimeOffset.UtcNow.AddSeconds(tokenResponse.ExpiresIn));
+        tx.Add(e, AuthInfo.SessionId, tokenResponse.SessionId);
+        tx.Add(e, AuthInfo.UserId, ulong.Parse(tokenResponse.UserId));
+        await tx.Commit();
+        _logger.LogInformation("Logged in successfully to GOG.");
+    }
+    
+
+    /// <summary>
+    /// Create a new HttpRequestMessage with the OAuth token.
+    /// </summary>
+    private async ValueTask<HttpRequestMessage> CreateMessage(Uri uri, CancellationToken token)
+    {
+        if (!TryGetAuthInfo(out var authInfo))
+            throw new InvalidOperationException("No authentication information found.");
+
+        if (NeedsRefresh(authInfo))
+            authInfo = await RefreshToken(token);
+        
+        var message = new HttpRequestMessage(HttpMethod.Get, uri);
+        message.Headers.Authorization = new("Bearer", authInfo.AccessToken);
+        return message;
+    }
+    
+    /// <summary>
+    /// Returns true if the OAuth token needs to be refreshed.
+    /// </summary>
+    private bool NeedsRefresh(AuthInfo.ReadOnly authInfo)
+    {
+        return authInfo.ExpiresAt < DateTimeOffset.UtcNow - TimeSpan.FromMinutes(1);
```

Comment:
Shouldn't this be
```suggestion
    private bool NeedsRefresh(AuthInfo.ReadOnly authInfo)
    {
        return authInfo.ExpiresAt < DateTimeOffset.UtcNow + TimeSpan.FromMinutes(1);
```

Assuming the idea was to make sure to refresh tokens before they expire